### ISSUE TITLE
Own messages sent from web client don’t show up in IRC

### ISSUE THREAD

**Issue by lucaswerkmeister** [gitterHQ/irc-bridge#5]
When I’m online on both IRC and the web client, messages I send from IRC show up in the web client, but messages I send from the web client don’t show up in IRC. (Messages from other people always show up in both clients.)

**Comment by malditogeek** [comment]
This is a limitation of the IRC design. When you send a message from the web client, the message is processed by the server and sent back to your browser you once we confirm everything is correct. On IRC when you send a message what you see as your own message on the interface is just a loopback handled by the IRC client itself, the server is not involved at all, even if the message is invalid and fails to be processed you'll still see the message the IRC client UI. Because of this is not possible to display as yours a message that was generated by yourself somewhere else.

This is why we have this rule here https://github.com/gitterHQ/irc-bridge/blob/master/lib/gitter-adapter.js#L105 , if that rule was not there you'd your messages duplicated in the IRC client UI.

Makes sense?

**Comment by lucaswerkmeister** [comment]
I don’t understand… if it’s “not possible to display as yours a message that was generated by yourself somewhere else”, then why do you need to explicitly filter out own messages with that rule? It seems to me that the existence of that rule implies that you *can* receive your own messages from the server.

In that case, AFAICT the IRC bridge would just need to tell where a message came from, and only filter out (in L105) those that came from the IRC client (to avoid the duplicate message). Is that possible?

**Comment by jimschubert** [comment]
The point is that gitter-adapter is just working with messages that have no metadata. Without the mentioned filter, you'd see every message twice in IRC. Some IRC clients may not even display a message 'received' from the server with your own username. I've never had that issue, though, using desktop and web interfaces to IRC.

In order to display messages in IRC that are sent from the web, the gitter client would have to include some source along with the message, something like:

```
if (nick === c.nick && evt.sourceInterface === 'IRC') return; // User's own IRC message
```

**Comment by lucaswerkmeister** [comment]
> `if (nick === c.nick && evt.sourceInterface === 'IRC') return; // User's own IRC message`

That’s what I was thinking as well… is it possible?

**Comment by malditogeek** [comment]
Yeah, this would be possible with message metadata. At the moment we don't store metadata on the message regarding where they generated but if we do in the future we could definitely do this.

**Comment by Kimundi** [comment]
Any update on this?
